src/test/kotlin/com/music/awesomemusic/controllers/SignUpTest.kt

```diff
@@ -0,0 +1,214 @@
+        headers.contentType = MediaType.APPLICATION_JSON
+        val request = HttpEntity<String>("{\"username\": \"test3\", \"password\": \"123HAha@%/+'!#$^?:.(){}[]~-_.\",\"email\":\"test3@mail.ru\",\"is_collective\": \"false\" }", headers)
+        val responseEntity = restTemplate.postForEntity("http://localhost:$port/api/user/registration", request, String::class.java)
+
+        assertEquals(responseEntity.statusCode, HttpStatus.CREATED)
+
+    }
+
+
+    @Test
+    @Throws(Exception::class)
+    fun shouldSignUpIfPasswordLess6Expected400() {
+
+        val headers = HttpHeaders()
+        headers.contentType = MediaType.APPLICATION_JSON
+
+        val request = HttpEntity<String>("{\"username\": \"viovio\", \"password\": \"12345\",\"email\":\"test3@mail.ru\",\"is_collective\": \"false\" }", headers)
+        val responseEntity = restTemplate.postForEntity("http://localhost:$port/api/user/registration", request, String::class.java)
+
+        assertEquals(responseEntity.statusCode, HttpStatus.BAD_REQUEST)
+    }
+
+
+    @Test
+    @Throws(Exception::class)
+    fun shouldSignUpIfPasswordMore32Expected400() {
+
+        val headers = HttpHeaders()
+        headers.contentType = MediaType.APPLICATION_JSON
+
+        val request = HttpEntity<String>("{\"username\": \"viovio\", \"password\": \"1234567890HaHaHaHAHAHAHAHAHAHAH%%%%%%%%%%%%%%%%\",\"email\":\"test3@mail.ru\",\"is_collective\": \"false\" }", headers)
+        val responseEntity = restTemplate.postForEntity("http://localhost:$port/api/user/registration", request, String::class.java)
+
+        assertEquals(responseEntity.statusCode, HttpStatus.BAD_REQUEST)
+    }
+
+
+    @Test
+    @Throws(Exception::class)
+    fun shouldSignUpIfEmailLess5Expected400() {
+
+        val headers = HttpHeaders()
+        headers.contentType = MediaType.APPLICATION_JSON
+
+        val request = HttpEntity<String>("{\"username\": \"Violetta\", \"password\": \"1233Agf%%%QWERTY%1qq\",\"email\":\"1@.r\",\"is_collective\": \"false\" }", headers)
+        val responseEntity = restTemplate.postForEntity("http://localhost:$port/api/user/registration", request, String::class.java)
+
+        assertEquals(responseEntity.statusCode, HttpStatus.BAD_REQUEST)
+    }
+
+    @Test
+    @Throws(Exception::class)
+    fun shouldSignUpIfEmailMore254Expected400() {
+
+        val headers = HttpHeaders()
+        headers.contentType = MediaType.APPLICATION_JSON
+
+        val request = HttpEntity<String>("{\"username\": \"Violetta\", \"password\": \"1233Agf%%%QWERTY%1qq" +
+                "qqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqq" +
+                "qqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqq\",\"email\":\"1@.r\",\"is_collective\": \"false\" }", headers)
```

Comment:
It's better to use some function that will generate a string for provided length.